### DISCUSSION TITLE
How can I create custom block with addon?

### DISCUSSION THREAD

**Discussion by GulgDev** [ScratchAddons/ScratchAddons#4961] [Help]
I want to create a group of new blocks? How can I do it? Should I use addon.tab.traps.getBlockly()?

**Comment by Platformer444** [comment]
You should use `addon.tab.addBlock`(see at https://scratchaddons.com/docs/reference/addon-api/addon.tab/#addontabaddblock). 

But the problem with this method is that it adds it only to the Debugger Category. But if you are adding the block to the Debugger Category then it isn't a problem.

**Comment by iqnite** [comment]
You could copy and modify the Debugger addon, but it's not easy.
Whatever you do, consider that custom blocks won't work for users without Scratch Addons, and they won't be accepted if they change a project's behavior on stage. If you want to expand Scratch block's functionality, you should make a [TurboWarp extension](https://extensions.turbowarp.org). The documentation for that is [here](https://docs.turbowarp.org/development/extensions/introduction).
